### PR TITLE
Small cleanups to a number of client behaviors

### PR DESCRIPTION
**Pull request by smarterclayton**
This is extracted from #5012 and contains small fixes and cleanups. The most
significant changes are allowing kubectl to properly handle objects that don't
exist in v1beta1/v1beta2

### CODE REVIEW

File: pkg/registry/generic/etcd/etcd.go [hunk 1]
```diff
@@ -327,11 +356,15 @@ func (e *Etcd) Delete(ctx api.Context, name string) (runtime.Object, error) {
 	return &api.Status{Status: api.StatusSuccess}, nil
 }
 
-// Watch starts a watch for the items that m matches.
+// WatchPredicate starts a watch for the items that m matches.
```
Comment (lavalamp) [inline pkg/registry/generic/etcd/etcd.go]: Nit: name doesn't match

**Comment by lavalamp** [comment]
LGTM. Nit is not a big deal. Should I ask for an e2e run, though? I think it's probably ok?

**Comment by smarterclayton** [comment]
I'll do an e2e.  It's somewhat big

----- Original Message -----

**Comment by smarterclayton** [comment]
Are you able to kick off an automated e2e on pulls through jenkins?  i'm having problems with my local e2e setup for some reason.

----- Original Message -----

**Comment by lavalamp** [comment]
Running e2e

**Comment by lavalamp** [comment]
OK guestbook failed for me but I'm pretty sure it's because of previous bad state in my e2e cluster.

LGTM

**Pull request merged**